### PR TITLE
Add SE050 test

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
This PR adds a test that is introduced in the admin app with https://github.com/Nitrokey/admin-app/pull/5

Note that it takes around 30seconds to complete, it's not crashing.

### CODE REVIEW

**Review by szszszsz — COMMENTED** [review]
I see that this is generally a caller for the in-firmware test, and interpreter of the results gathered there.
1. Why the control of the tests is placed in the firmware instead of the test calling code?
Was it easier that way? Or is this a quick POC?
2. I think `pytest` would be better to use here, but its done already.

**Comment by sosthene-nitrokey** [comment]
> You are essentially calling a fixed test routine in the firmware, and waiting for the results. Instead you could construct test cases on the calling side, and use a proper direct communication channel to se050.

I wanted to test the actual APDU generation from the driver to test it, so all commands to the SE050 needs to be run from rust code. This is meant to test the driver in real conditions. Communicating to the SE050 directly from nitropy would make that much less useful, even if it were to test the T=1 over I2C part of the driver.

> I do not think though running pytest itself is that much trouble for a beta tester, but that's my opinion.

I agree, but running pytest is much less discoverable while `nk3 test` is documented and used.

File: pynitrokey/cli/nk3/test.py [hunk 1]
```diff
@@ -233,6 +237,191 @@ def select(conn: CardConnection, aid: list[int]) -> bool:
...
+    bar = tqdm(
+        desc="Running SE050 test", unit="%", bar_format="{l_bar}{bar}", total=900
+    )
+    # 1m30 in increments of 0.1 second
+    for i in range(900):
+        t.join(0.1)
+        bar.update(1)
+        if not t.is_alive():
+            break
+        if i == 900 - 1:
```
Comment (robin-nitrokey) [inline pynitrokey/cli/nk3/test.py:386]: nit: use a constant for the timeout

**Review by robin-nitrokey — APPROVED** [review]

File: pynitrokey/cli/nk3/test.py (hunk 1 above)
Comment (robin-nitrokey) [reply pynitrokey/cli/nk3/test.py:386]: and maybe use `else` for the last occurence

Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py]: for-else is nice

**Review by daringer — APPROVED** [review]

**Comment by daringer** [comment]
this does not interfere with anything else, should we merge this ?

**Comment by robin-nitrokey** [comment]
Running this with the `se050-test-app` feature disabled now gives me:
```
[4/5]	se050    	SE050                   	FAILURE  	object of type 'type' has no len()
```

File: pynitrokey/cli/nk3/test.py [hunk 2]
```diff
@@ -368,10 +373,18 @@ def test_se050(ctx: TestContext, device: Nitrokey3Base) -> TestResult:
     ):
         return TestResult(TestStatus.SKIPPED)
 
-    que: Queue[Optional[bytes]] = Queue()
+    que: Queue[Union[bytes, None, type[NotSupported]]] = Queue()
 
-    def internal_se050_run(q: Queue[Optional[bytes]]) -> None:
-        q.put(AdminApp(device).se050_tests())
+    def internal_se050_run(
+        q: Queue[Union[bytes, None, type[NotSupported]]],
```
Comment (robin-nitrokey) [inline pynitrokey/cli/nk3/test.py:379]: should be `NotSupported()`

File: pynitrokey/cli/nk3/test.py [hunk 3]
```diff
@@ -368,10 +373,18 @@ def test_se050(ctx: TestContext, device: Nitrokey3Base) -> TestResult:
...
+    ) -> None:
+        try:
+            q.put(AdminApp(device).se050_tests())
+        except CtapError as e:
+            if e.code == CtapError.ERR.INVALID_LENGTH:
+                q.put(NotSupported)
```
Comment (robin-nitrokey) [inline pynitrokey/cli/nk3/test.py:385]: should be `NotSupported()`

**Comment by sosthene-nitrokey** [comment]
> Running this with the `se050-test-app` feature disabled now gives me:
> 
> ```
> [4/5]	se050    	SE050                   	FAILURE  	object of type 'type' has no len()
> ```


Can you try again? This should be fixed I believe.

File: pynitrokey/cli/nk3/test.py (hunk 3 above)
Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py]: Fixed

**Comment by robin-nitrokey** [comment]
Yes, fixed.

File: pynitrokey/cli/nk3/test.py [hunk 4]
```diff
@@ -234,6 +239,213 @@ def select(conn: CardConnection, aid: list[int]) -> bool:
...
+
+    que: Queue[Union[bytes, None, NotSupported]] = Queue()
+
+    def internal_se050_run(
+        q: Queue[Union[bytes, None, NotSupported]],
+    ) -> None:
+        try:
+            q.put(AdminApp(device).se050_tests())
+        except CtapError as e:
+            if e.code == CtapError.ERR.INVALID_LENGTH:
```
Comment (robin-nitrokey) [inline pynitrokey/cli/nk3/test.py:384]: We could also check for `INVALID_COMMAND` instead of whitelisting firmware versions.

Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py:384]: Returning `UnsupportedCommand` seems to lead to pynitrokey not receiving the error.

Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py:384]: Do you know why that is there: https://github.com/Nitrokey/pynitrokey/blob/master/pynitrokey/nk3/admin_app.py#L72

Comment (robin-nitrokey) [reply pynitrokey/cli/nk3/test.py:384]: I think the idea was to make it easier to handle the case that the firmware does not (yet) support a command.  `_call` always returns `bytes`, so `None` always indicates that the command is not supported.  And checking for `None` is easier than catching the exception and checking the error code.

Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py:384]: Ok, so I can just skip if None.

Comment (sosthene-nitrokey) [reply pynitrokey/cli/nk3/test.py:384]: Relevant PR in the admin app: https://github.com/Nitrokey/admin-app/pull/12

**Pull request merged**